feat(documentation): documentation contribution add more details

LilyYin17 opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Closes #XXXXX

<!-- Feel free to add any additional description of changes below this line -->

LilyYin17 committed:
feat(documentation): add some words to specify for users and developers

LilyYin17 committed:
feat(documentation): add some words to specify for users and developers

LilyYin17 committed:
feat(documentation): add more details for contribution choices

LilyYin17 committed:
feat(documentation): add more details for contribution choices

samarmohan commented on README.md:
```diff
@@ -15,10 +15,10 @@ Our full-stack web development and machine learning curriculum is completely fre
 
 - [Certifications](#certifications)
-- [The Learning Platform](#the-learning-platform)
+- [The Learning Platform - for users](#the-learning-platform)
```
It is definitely possible that developers would want to learn and normal users would want to contribute. Normal users could contribute to documentation and devs are always leveling up their skills.

Sembauke replied:
This seems like an unnecessary change, and I agree with Samarmohan.

samarmohan commented on README.md:
```diff
@@ -172,9 +172,14 @@ We appreciate any responsible disclosure of vulnerabilities that might impact th
 ### Contributing
 
-The freeCodeCamp.org community is possible thanks to thousands of kind volunteers like you. We welcome all contributions to the community and are excited to welcome you aboard.
+The freeCodeCamp.org community is possible thanks to thousands of kind volunteers like you. We welcome all contributions to the community and are excited to welcome you aboard. Here is an overview of what you can contribute:
```
`Here is an overview of what you can contribute:` should be `Here is an overview of what you can do to help:` This is clearer, cleaner, and flows better!

Sembauke replied:
As stated above this should be changed.

moT01 replied:
What about `Here are some ways you can contribute:`

samarmohan commented on README.md:
```diff
@@ -172,9 +172,14 @@ We appreciate any responsible disclosure of vulnerabilities that might impact th
 
-The freeCodeCamp.org community is possible thanks to thousands of kind volunteers like you. We welcome all contributions to the community and are excited to welcome you aboard.
+The freeCodeCamp.org community is possible thanks to thousands of kind volunteers like you. We welcome all contributions to the community and are excited to welcome you aboard. Here is an overview of what you can contribute:
+- Answering coding questions on our community forum.
```
Fix tenses: `Answering` -> `Answer`

Sembauke replied:
Same here.

samarmohan commented on README.md:
```diff
@@ -172,9 +172,14 @@ We appreciate any responsible disclosure of vulnerabilities that might impact th
+- See the [issues](https://github.com/freeCodeCamp/freeCodeCamp/issues) for a list of proposed features (and known issues).
 
-> #### [Please follow these steps to contribute](https://contribute.freecodecamp.org).
+> #### [For more details, please follow these steps to contribute](https://contribute.freecodecamp.org).
```
The part before the comma doesn't flow or relate to the part after. I think you should keep it the same.

Sembauke replied:
I think this should not be changed, we want the contributors to read the contributing guide before contributing. This change seems to make it more optional.

samarmohan commented:
Sorry if this sounded harsh, thank you so much though!

Sembauke commented:
Hey @LilyYin17,

Thanks for making these changes, they look great!

But can you apply the above changes suggested?

Have a nice day and happy coding.

moT01 committed:
fix(curriculum): allow any response status for error response (#41906)

* fix(curriculum): allow any response status for error response

* Update curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/timestamp-microservice.md

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

* Update curriculum/challenges/english/05-apis-and-microservices/apis-and-microservices-projects/timestamp-microservice.md

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

Co-authored-by: Oliver Eyton-Williams <ojeytonwilliams@gmail.com>

gikf committed:
fix(curriculum): rework Project Euler 38 (#42321)

* fix: rework challenge to use argument in function

* fix: update solution to challenge changes

* fix: use MathJax to improve look of math notation

ojeytonwilliams committed:
feat: make api only return json (#42320)

* fix(api): return json for delete + reset-progress

* refactor(api): make .json calls explicit

These .send() calls had objects as arguments. Using .json() makes it
explicit without changing behaviour.

* fix: return json or redirectWithFlash

We should never be getting requests for plain text, but if we do they
should be redirected back to the client.

* fix: prioritize JSON responses

If accepted, respond with JSON.  If not, it's probably a bad request.

LilyYin17 committed:
feat(documentation): add whay you can contribute section

LilyYin17 committed:
feat(doumentation): word improving for contributing section

LilyYin17 commented:
Did the revisions based on your suggestions, and pull the latest code.

ShaunSHamilton commented:
@LilyYin17 Thank you, for making those changes. However, you appear to have pushed changes from other commits, as opposed to rebasing this branch with `main`. Could you please `revert` the last few commits to this branch?

nhcarrigan commented:
I'm not sure these changes are needed. The README currently links to the contributing documentation, where the landing page has the list this PR adds with the relevant links to the documentation.

Rather than linking to the GitHub issues tab in the readme, I think the current flow of encouraging contributors to start with the docs is ideal.

ShaunSHamilton commented:
@nhcarrigan Makes a decent point. After the recent suggestions, this PR adds information already on the main - what I call - _contributors page_ - where we actually want contributors to go.

As it stands, I find the amount of _stuff_ in the README too much.

Thank you, for all your time on this, and sorry about the to-and-fro, but I think we are going to close this PR.
